mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -184,21 +184,20 @@ class LogisticRegression(override val uid: String)
     }
 
     val states = optimizer.iterations(new CachedDiffFunction(costFun),
-      initialWeightsWithIntercept.toBreeze.toDenseVector)
+      initialWeightsWithIntercept.toBreeze.toDenseVector).toArray
```

Comment:
This will materialize all of the states, which could be pretty huge (since State stores vectors of weights, gradients, etc.).  Should you keep your original loop?